### PR TITLE
Show rotation feedback when editing PhysX collider

### PR DESCRIPTION
**Pull request by guillaume-haerinck**
Signed-off-by: guillaume-haerinck <guillaume.haerinck@outlook.com>

## What does this PR do?

When editing a PhysX collider, show the rotation feedback (as done when rotating object).
Suggested by @hultonha on https://github.com/o3de/o3de/pull/11550.

**Changes**

1. Call `DisplayFeedback` in the `ColliderRotationFeedback`
2. CircleViewFeedback is no longer a public member of `RotationManipulators`. Its updates are now handled inside the class by adding hook into the inputs callbacks
3. Subscribe to input callback in the `RotationManipulators` constructor, as when one is missing there can be visual glitches in the rotation feedback
4. Move `RotationManipulators::m_viewAngularManipulator` init in the class initialization list because we can

**In action**

Before :

https://user-images.githubusercontent.com/19243508/187020032-0840cf48-76e7-4564-b0d6-c7a29927e5fe.mp4

After :

https://user-images.githubusercontent.com/19243508/187020039-210e4ca4-4efc-47f3-9825-cec9aba987af.mp4

## How was this PR tested?

Rotate multiple physX collider types in subedit, and checked against object rotation to ensure that nothing broke.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp [hunk 1]
```diff
@@ -13,45 +13,74 @@
 namespace AzToolsFramework
 {
     RotationManipulators::RotationManipulators(const AZ::Transform& worldFromLocal)
+        : m_viewAngularManipulator { AngularManipulator::MakeShared(worldFromLocal) }
     {
+        const auto saveMostRecentActionCb = [this](const AngularManipulator::Action& action)
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp:18]: nit: This is very minor but in manipulator code (AzToolsFramework) we tend to use `Fn` postfix for lambda variables as opposed to `Cb` but it's not a huge deal, just making you aware 🙂 (can change or not, not blocking)

**Review by hultonha — APPROVED** [review]
Really lovely refactor here, this is great that we contain the feedback in the rotation manipulator class using the extra lambda indirection 🙂 Very cool to see!

**Comment by hultonha** [comment]
Just kicked off a build now too - https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-11606/1/

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp [hunk 2]
```diff
@@ -13,45 +13,74 @@
...
+        const auto saveMostRecentActionHookCb = [this, onMouseDownCallback](const AngularManipulator::Action& action)
+        {
+            m_angularManipulatorFeedback.m_mostRecentAction = action;
+            onMouseDownCallback(action);
+        };
+
         for (AZStd::shared_ptr<AngularManipulator>& manipulator : m_localAngularManipulators)
         {
-            manipulator->InstallLeftMouseDownCallback(onMouseDownCallback);
+            manipulator->InstallLeftMouseDownCallback(saveMostRecentActionHookCb);
```
Comment (greerdv) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp:48]: Nitpick - maybe it's just me, but this naming seems a tiny bit unclear. The callback is doing two things - saving the most recent action, and then doing the mouse down behaviour, but the name only communicates the first of those things to me, and kind of implies that saving the action is the most important thing that's happening here. I'd also say it's a little bit confusing to use the same name for other lambdas later on which do different things.

**Review by greerdv — APPROVED** [review]
Looks good to me. Great change, thanks for doing this!

**Review by jonawals — APPROVED** [review]
LGTM

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp (hunk 1 above)
Comment (guillaume-haerinck) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp:18]: Ok will do :)

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp (hunk 2 above)
Comment (guillaume-haerinck) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/RotationManipulators.cpp:48]: Ok I'll make it more descriptive 👌

**Pull request merged**